fix(ci): api-contracts.typecheck.ts fails — internal SessionInfo missing API contract fields

## Summary

`api-contracts.typecheck.ts` fails with 3 errors on fresh CI runs (passes intermittently due to caching):

```
src/api-contracts.typecheck.ts(20,48): error TS2344: Type 'false' does not satisfy the constraint 'true'.
src/api-contracts.typecheck.ts(22,3): error TS2344: Type 'false' does not satisfy the constraint 'true'.
src/api-contracts.typecheck.ts(25,3): error TS2344: Type 'false' does not satisfy the constraint 'true'.
```

The typecheck asserts that `InternalSessionInfo extends SessionInfo` (from `api-contracts.ts`). The API contract has fields added in recent PRs that the internal type doesn't declare:

- `autoApprove?: boolean`
- `pendingPermission?: PendingPermissionInfo`
- `pendingQuestion?: PendingQuestionInfo`
- `promptDelivery?: { delivered: boolean; attempts: number }`
- `actionHints?: Record<string, { method: string; url: string; description: string }>`

## Fix
Either:
1. Add the missing fields to internal `SessionInfo` in `src/session.ts`
2. Or regenerate the OpenAPI contract via `npm run build:openapi`

## Impact
Blocks all PRs on CI (sdk-drift, test, platform-smoke all fail at tsc step).

## Found by
PR #2369 CI failure (also blocking other in-flight PRs)

--- a/src/session.ts
+++ b/src/session.ts
@@ -30,7 +30,7 @@ import { Mutex } from 'async-mutex';
 import { maybeInjectFault } from './fault-injection.js';
 import { startSessionSpan, startTmuxSpan, spanError, spanOk } from './tracing.js';
 import type { Span } from '@opentelemetry/api';
-import type { PendingPermissionInfo } from './api-contracts.js';
+import type { PendingPermissionInfo, PendingQuestionInfo } from './api-contracts.js';
 
 /** Convert parsed JSON arrays to Sets for activeSubagents (#668). */
 // Cache for hook cleanup to avoid running on every createSession (Issue #1134).
@@ -113,6 +113,11 @@ export interface SessionInfo {
   prd?: string;                // Issue #735: Optional PRD contract text attached to the session
   ownerKeyId?: string;         // Issue #1429: API key ID that created this session (ownership)
   tenantId?: string;           // Issue #1944: Tenant isolation scoping
+  autoApprove?: boolean;        // API contract compat: auto-approve flag
+  pendingPermission?: PendingPermissionInfo;  // API contract compat: active permission prompt
+  pendingQuestion?: PendingQuestionInfo;       // API contract compat: active question
+  promptDelivery?: { delivered: boolean; attempts: number };  // API contract compat: prompt status
+  actionHints?: Record<string, { method: string; url: string; description: string }>;  // API contract compat: actionable hints
 }
 
 /** Persisted session store keyed by Aegis session ID. */

